compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1143,8 +1150,12 @@ trait Applications extends Compatibility {
     val typedArgs = if (isNamed) typedNamedArgs(tree.args) else tree.args.mapconserve(typedType(_))
     record("typedTypeApply")
     typedExpr(tree.fun, PolyProto(typedArgs, pt)) match {
-      case _: TypeApply if !ctx.isAfterTyper =>
-        errorTree(tree, em"illegal repeated type application")
+      case fun: TypeApply if !ctx.isAfterTyper =>
+        val function = fun.fun
+        val args = (tree.args ++ fun.args).map(_.show).mkString(", ")
```

Comment:
Shouldn't it be the other way around?
```suggestion
        val args = (fun.args ++ tree.args).map(_.show).mkString(", ")
```
Is there a test for this?